pkgs/development/libraries/gdal/tests.nix

```diff
@@ -13,6 +13,16 @@ runCommand "${pname}-tests" { meta.timeout = 60; }
     ${gdal}/bin/gdalinfo --version \
       | grep 'GDAL ${version}'
 
+    # test java bindings
+    cat <<EOF > main.java
+    import org.gdal.gdal.gdal;
+    class Main {
+      public static void main(String[] args) {
+      gdal.AllRegister();
+      }
+    }
+    EOF
+    ${jdk}/bin/java -Djava.library.path=${gdal}/lib/ -cp ${gdal}/share/java/gdal-${version}.jar main.java
```

Comment:
This example is based on https://gdal.org/api/java/index.html#how-to-use-the-bindings.